generator/src/main/java/org/corfudb/generator/operations/WriteOperation.java

```diff
@@ -1,39 +1,75 @@
 package org.corfudb.generator.operations;
 
+import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
-import org.corfudb.generator.Correctness;
-import org.corfudb.generator.State;
-import org.corfudb.runtime.CorfuRuntime;
-import org.corfudb.runtime.object.transactions.TransactionalContext;
+import org.corfudb.generator.correctness.Correctness;
+import org.corfudb.generator.distributions.Keys;
+import org.corfudb.generator.state.CorfuTablesGenerator;
+import org.corfudb.generator.state.KeysState;
+import org.corfudb.generator.state.KeysState.ThreadName;
+import org.corfudb.generator.state.State;
 
+import java.util.Optional;
 import java.util.UUID;
 
 /**
- * Created by maithem on 7/14/17.
+ * Writes a new value to the corfu table
  */
 @Slf4j
 public class WriteOperation extends Operation {
+    @Getter
+    private final Context context;
+    private final CorfuTablesGenerator tableManager;
+    private final Correctness correctness;
 
-    public WriteOperation(State state) {
-        super(state, "Write");
+    public WriteOperation(State state, CorfuTablesGenerator tableManager, Correctness correctness) {
+        super(state, Operation.Type.WRITE);
+        this.tableManager = tableManager;
+        this.correctness = correctness;
+
+        Keys.FullyQualifiedKey key = generateFqKey(state);
+
+        this.context = Context.builder()
+                .fqKey(key)
+                .val(Optional.of(UUID.randomUUID().toString()))
+                .build();
+    }
+
+    @Override
+    public boolean verify() {
+        return false;
```

Comment:
Does this mean that If a put is generated, but that key is never read then that write won't be verified?

Reply:
WriteOperation is not used in the injection framework, I see it in code and I double-checked it in logs. There are only transactional writes

Reply:
In current java implementation, if WriteOperation would be run accidentally the the entire verification will fail

Reply:
Non transactional writes are not generated, but this verification also needs to happen for transaction writes. For example, txn.begin | write k1,v1 | txn.end , if a read operation to k1 is never generated then it won't be verified that it has indeed been written to the db.

Reply:
WriteOperation:verify() will never be executed, since any write is under a transaction then only "WriteAfterWriteTxOperation#verify" will be executed always.

Anyways, I have removed all verify methods, to reduce the complexity of this PR, new there are only refactorings that prepares longevity to being updated with "verify" functionality